mcux/middleware/wifi_nxp/wifidriver/wifi.c

```diff
@@ -0,0 +1,5239 @@
+/** @file wifi.c
+ *
+ *  @brief  This file provides WiFi Core API
+ *
+ *  Copyright 2008-2024 NXP
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+#include <mlan_api.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <wifi.h>
+#include <osa.h>
+
+#include "wifi-internal.h"
+#include <wm_net.h>
+#if defined(RW610)
+#include "wifi-imu.h"
+#else
+#include <mlan_sdio_api.h>
+#include "wifi-sdio.h"
+#include "mlan_sdio.h"
+#include "sdio.h"
+#include "firmware_dnld.h"
+
+#endif
+#ifdef RW610
+#include "fsl_adapter_rfimu.h"
+#endif
+
+/* Always keep this include at the end of all include files */
+#include <mlan_remap_mem_operations.h>
+
+#if CONFIG_HEAP_DEBUG
+OSA_SEMAPHORE_HANDLE_DEFINE(os_mem_stat_sem);
+
+t_u32 valid_item_cnt = 0;
+wifi_os_mem_info wifi_os_mem_stat[OS_MEM_STAT_TABLE_SIZE];
+#endif
+
+#if CONFIG_CSI
+#define MAX_CSI_LOCAL_BUF        80
+#define CSI_LOCAL_BUF_ENTRY_SIZE 768
+t_u8 csi_local_buff[MAX_CSI_LOCAL_BUF][CSI_LOCAL_BUF_ENTRY_SIZE] = {
+    0,
+};
+
+csi_local_buff_statu csi_buff_stat = {0, 0, 0};
+
+int csi_event_cnt        = 0;
+t_u64 csi_event_data_len = 0;
+#endif
+
+#if CONFIG_ECSA
+extern wifi_ecsa_status_control ecsa_status_control;
+#endif
+
+#if !CONFIG_WIFI_CORE_STACK_SIZE
+#define CONFIG_WIFI_CORE_STACK_SIZE (2048)
+#endif
+
+#define MAX_MCAST_LEN (MLAN_MAX_MULTICAST_LIST_SIZE * MLAN_MAC_ADDR_LENGTH)
+#if CONFIG_WiFi_878x
+#define MAX_WAIT_TIME 20
+#else
+#define MAX_WAIT_TIME 3000
+#endif
+
+#ifndef USB_SUPPORT_ENABLE
+#define _T(x) x
+#endif
+
+#if CONFIG_WMM
+#define BOARD_DATA_BUFFER_ALIGN_SIZE 32
+
+SDK_ALIGN(uint8_t outbuf_arr[MAX_WMM_BUF_NUM][OUTBUF_WMM_LEN], BOARD_DATA_BUFFER_ALIGN_SIZE);
```

Comment:
Ping @MaochenWang1- these definitions need to be fixed- could we create a definition like `WL_ALIGN`?